app/src/main/java/com/example/hungrywolfes/ui/detailsScreen/DetailsViewModel.kt

```diff
@@ -36,11 +39,59 @@ class DetailsViewModel : ViewModel() {
             try {
                 _detailsMeal.value =
                     CategoryApi.retrofitService.detailsFood(item).meals.firstOrNull()
+                buttonCheckStatus.value = itemInFavorite()
             } catch (e: java.lang.Exception) {
             }
         }
     }
+
     fun onFavoriteButton() {
-        _addItemToFavoriteButton.value = addItemToFavoriteButton.value == false
+        if (itemInFavorite()) {
+            buttonCheckStatus.value = false
+            removeMealFromFavorite()
+
+
+        } else {
+            addMealToFavorite()
+            buttonCheckStatus.value = true
+        }
+    }
+
+    private fun addMealToFavorite() {
+        detailsMeal.value?.let { details ->
+            favoriteMealList.add(
+                ListMealsImages(
+                    details.strMeal,
+                    details.strMealThumb,
+                    details.idMeal
+                )
+            )
+        }
+        Hawk.put(keyHawk, favoriteMealList)
+    }
+
+    private fun removeMealFromFavorite() {
+        detailsMeal.value?.let { details ->
+            favoriteMealList.remove(
+                ListMealsImages(
+                    details.strMeal,
+                    details.strMealThumb,
+                    details.idMeal
+                )
+            )
+        }
+        Hawk.put(keyHawk, favoriteMealList)
+    }
+
+    private fun itemInFavorite(): Boolean {
+        return detailsMeal.value?.let { details ->
+            favoriteMealList.contains(
```

Comment:
Here instead of creating a new object, you could use `.any` to check by id if there is such an item
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/any.html